src/metabase/query_processor/util/nest_query.clj

```diff
@@ -135,6 +149,9 @@
                            desired-alias (assoc ::add/source-alias desired-alias
                                                 ::add/desired-alias desired-alias))])
 
+    [:field (id-or-name :guard string?) (opts :guard ::add/source-alias)]
```

Comment:
is this making a nominal-ref? does similar need to be done for `:expression`? 
a comment for this condition would be helpful

Reply:
Not exactly. It's rewriting from naive `:name` to possibly adjusted source alias, for any nominal references in the outer layer of the query. There were cases where this was generating `"source"."ID"` when the column was really called `"source"."People - User__ID"`.

Regarding `:expression`, no, because they're in different stages so there wouldn't be an `:expression` ref, it would be a `:field` ref.

I'll add a comment explaining about this.

Reply:
Done.